src/main/java/org/magicdgs/popgenlib/neutralitytest/TajimasD.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2017 Daniel Gomez-Sanchez
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.magicdgs.popgenlib.neutralitytest;
+
+import org.magicdgs.popgenlib.diversity.NucleotideDiversity;
+import org.magicdgs.popgenlib.utils.Verify;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.math3.util.FastMath;
+import org.apache.commons.math3.util.Pair;
+
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+
+/**
+ * Static methods to compute Tajima's D.
+ *
+ * <p>REFERENCES:
+ * <ul>
+ *
+ * <li><a href="http://www.genetics.org/content/123/3/585">
+ * Tajima (1989): Statistical method for testing the neutral mutation hypothesis by DNA
+ * polymorphism, Genetics 123(3).
+ * </a></li>
+ *
+ * </ul>
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public final class TajimasD {
+
+    // cannot be instantiated
+    private TajimasD() {}
+
+    /**
+     * Computes Tajima's D using sample summaries using the formula 38 in
+     * <a href="http://www.genetics.org/content/123/3/585">Tajima (1989)</a>
+     *
+     * @param numberOfSamples          number of samples used to get the number of segregating
+     *                                 sites.
+     * @param numberOfSegregatingSites number of segregating sites in the sample.
+     * @param pairwiseDifferences      pair-wise differences between samples.
+     */
+    public static double tajimasD(final int numberOfSamples, final int numberOfSegregatingSites,
+            final double pairwiseDifferences) {
+        Verify.validate(numberOfSegregatingSites >= 0,
```

Comment:
True, and actually this is also an error in NucleotideDiversity.wattersonsTheta(). Changed here and added issue to for the other case...